src/core/drive/navigator.ts

```diff
@@ -119,6 +119,11 @@ export class Navigator {
     this.delegate.visitCompleted(visit)
   }
 
+  locationWithActionIsSamePage(location: URL, action: Action): boolean {
+    return getRequestURL(location) === getRequestURL(this.view.lastRenderedLocation) &&
+      (getAnchor(location) != null || action == "restore")
```

Comment:
Can you elaborate on what situation the `|| action == "restore"` condition is handling? I don't think this should ever affect the native apps, since restores are initiated by the apps themselves, but want to make sure.

Reply:
@jayohms sure, this refers to the changes in: https://github.com/hotwired/turbo/pull/298/commits/0800ec84f9e79a02af6c4d5592274dec7179bf31 It prevents unnecessary requests when navigating back _from_ a same-page anchor to a location with no anchor, when caching is disabled or has been cleared. e.g.

1. Visit `example.com` which has Turbo caching disabled
2. Click link to `example.com#main`
3. Tap Back

Without this change, Turbo would re-request `example.com` because the anchor is `undefined`. If the request URLs are equal, and the action is `restore`, we can be sure that the visit is a same-page one.

Reply:
Perfect, makes sense 👍